doc/api/http.md

```diff
@@ -490,27 +498,29 @@ Once a socket is assigned to this request and is connected
 added: v0.5.9
 -->
 
-Once a socket is assigned to this request and is connected
-[`socket.setTimeout()`][] will be called.
-
 * `timeout` {Number} Milliseconds before a request is considered to be timed out.
 * `callback` {Function} Optional function to be called when a timeout occurs. Same as binding to the `timeout` event.
 
+Once a socket is assigned to this request and is connected
```

Comment:
This is not 100% true, I stumbled upon this the other day. There are [cases](https://github.com/nodejs/node/blob/d196c5d2af6b5727289bf2da68514517bd1c257f/lib/_http_client.js#L635-L637) where the `socket.setTimeout()` is called even if the socket is not connected.
Anyway it's probably better to investigate/fix this in another issue/PR.